.github/PULL_REQUEST_TEMPLATE/release_checklist.md

```diff
@@ -0,0 +1,35 @@
+- [ ] [Create blog announcement PR](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#submit-a-release-notes-blog-post-pr) and solicit team member input
+- [ ] AppVeyor run success
+- [ ] [Buildkite](https://buildkite.com/chef/habitat-sh-habitat-master-release) run success
+- [ ] [Validate](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#validate-the-release) [darwin binaries](https://bintray.com/habitat/stable/hab-x86_64-darwin)
+- [ ] [Validate](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#validate-the-release) [linux binaries](https://bintray.com/habitat/stable/hab-x86_64-linux)
+- [ ] [Validate](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#validate-the-release) [windows binaries](https://bintray.com/habitat/stable/hab-x86_64-windows)
+- [ ] Declare merge thaw and update slack status
+- [ ] Merge blog announcement PR
+- [ ] [Update builder bootstrap bundle](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#update-builder-bootstrap-bundle)
+- [ ] [Update homebrew tap](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#update-homebrew-tap)
+- [ ] [Update chocolately package](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#rerun-chocolatey-validation-tests)
+- [ ] [Publish release](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#publish-release) on GitHub
+- [ ] [Update `hab-backline` to release version in acceptance environment](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#update-the-acceptance-environment-with-the-new-hab-backline)
+- [ ] [Update docs](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#update-the-docs)
+- [ ] [Bump version](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#bump-version) to new `-dev` version
+- [ ] [Update `hab-backline` to new `-dev` version in acceptance environment](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#update-the-acceptance-environment-with-the-new-hab-backline-1)
+- [ ] [Promote builder worker](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#promote-the-builder-worker) and confrm build uses new version
+- [ ] Post announcement in [Chef discourse](https://discourse.chef.io/c/habitat)
+- [ ] Post announcement in [Habitat forums](https://forums.habitat.sh/c/announcements)
+- [ ] Tweet announcement from `@habitatsh`
+- [ ] [Update `Cargo.lock`](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#update-cargolock) for [`habitat`](https://github.com/habitat-sh/habitat)
+- [ ] [Update `Cargo.lock`](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#update-cargolock) for [`core`](https://github.com/habitat-sh/core)
+- [ ] [Update `Cargo.lock`](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md#update-cargolock) for [`builder`](https://github.com/habitat-sh/builder)
+- [ ] Add fixes to the release automation and/or [`RELEASE.md`](https://github.com/habitat-sh/habitat/blob/master/RELEASE.md)
+- [ ] Merge this PR
```

Comment:
What are the concrete items that are expected to be included in this PR? From my reading, it might end up looking like this:
- [ ] documentation updates
- [ ] version bump
- [ ] `Cargo.lock` updates for this repository
- [ ] Any needed updates to `RELEASE.md` (and this template, presumably)

I've generally included the blog post in this "post-release" PR, but I can see an argument for getting that announcement out sooner, given the other things to be done here.

I've also generally done the `Cargo.lock` updates as their own thing, just to keep that change more isolated, and also more in line with the corresponding updates in `builder` and `core`.

If the version bump is supposed to be in this PR, I'd argue that the merge freeze should not be lifted until it is merged, to ensure that any subsequently-built packages would be on the new `*-dev` line, rather than on the just-released line. Given the other things that are in this PR, however, that may take a while to happen.

If the version bump is supposed to go in with its own PR, in order to lift the merge freeze as soon as possible, we should explicitly call that out in the checklist. 

Given all that, what should the expected contents of this PR we're merging be?

Reply:
The only thing I'd want to see in this PR is updates (code and/or `RELEASE.md`) for the next release process.

- The (auto generate) documentation update PR (line 24) should be pushed and validated before the release is considered done, but if you'd rather include is as part of this PR's content, I don't have terribly strong feelings.
- The version bump needs to happen before `Update `hab-backline` to new `-dev` version` (line 26)
- `Cargo.lock` updates don't really have anything to do with the release checklist, so I'd prefer they be separate PRs.

Reply:
My general concern is that it's not clear from the text of the checklist exactly what is supposed to be in _this_ PR, (or that it's indeed possible and normal for there not to be _any_ PR that results!), and which steps generate their own PRs. Granted, the linked `RELEASE.md` sections _do_ say when to make a PR, but it's not obvious looking at this, which I assume is to be the main driver of the manual process.

Reply:
Added [some additional explanation](https://github.com/habitat-sh/habitat/pull/5845/commits/669abbd3e45401c28287bef8b9dfeb430d329f99). Does that work?